software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -112,48 +106,65 @@ public void install() {
                 .execute();
     }
 
-    private List<String> installLinux(String expandedInstallDir) {
```

Comment:
can we keep the old code -- maybe a new static `newTaskInstallToLinuxFromFiles(...)` method, possibly refactored to a `RiakInstallationUtils` class?

ideally this would build up a task tree as part of the process -- using `Tasks` and `DynamicTasks` -- rather than create an enormous script.  i.e. trying the `packagecloud.io` install first, and if that fails it trying the old-style `fromFiles` link instead.  in the process @bostko you will probably encounter plenty of questions on tasks.  ask on these on IRC, and if necessary put together `tasks.md` page for the documentation.  (this is a bigger piece of work than is necessary, but it's something we've wanted for a while, plus it should be more interesting for you!)